### PR TITLE
Convert cppException unittest to edger8r

### PR DESCRIPTION
**Pull request by EMumau**

### CODE REVIEW

File: tests/cppException/enc/enc.cpp [hunk 1]
```diff
@@ -5,47 +5,31 @@
 #include <openenclave/enclave.h>
 #include <openenclave/internal/atexit.h>
 #include <openenclave/internal/print.h>
-#include "../args.h"
+#include "cppException_t.h"
 
 bool TestCppException();
 
-OE_ECALL void Test(void* args_)
+extern "C" int test(void)
```
Comment (anitagov) [inline tests/cppException/enc/enc.cpp]: Can you retain the function name as Test unless this is forbidden by the naming convention?

**Comment by anitagov** [comment]
How are the results of these changes with CI?

File: tests/cppException/enc/enc.cpp (hunk 1 above)
Comment (EMumau) [reply tests/cppException/enc/enc.cpp]: The Development Guide states that the Naming Convention is lower_camel_case.  I have used the lower_camel_case naming convention in the EDL file in order to comply with the naming convention.

Comment (anitagov) [reply tests/cppException/enc/enc.cpp]: Thanks.

**Comment by anitagov** [comment]
@EMumau - It is correct practice to run bors try before you do a bors r+. There were no bors try ever run on this issue. Doing a sandbox run will exercise your branch but without getting latest changes to master.
@andschwa @johnkord - Correct me if I am wrong here.

**Comment by johnkord** [comment]
@anitagov @EMumau We haven't established any rules about "bors try" being required (yet). Something like that will come soon though!

**Comment by EMumau** [comment]
@anitagov @johnkord  I could be mistaken, but I believe @andschwa explained to me that "bors try" is unnecessary extra step.  That "bors r+" executes the same validation.

**Comment by johnkord** [comment]
@EMumau It's true usually that it's an unnecessary extra step, but we hit an issue on Friday that I think justifies changing those guidelines for the future.  I think in the future we'll go with a rule like: "if the PR is changing the CI system (like modifying the Jenkinsfile or test scripts), then a 'bors try' is required to be run and its output manually looked at by reviewers before running a 'bors r+' to merge to master, in order to make sure that build/test steps are indeed being run"

The reason for this proposed change is that last week our CI system had been failing on Windows, but reporting that all builds were successful on Windows, and none of us knew until Friday that it had been doing this because we merged the changes that caused this to happen without manually verifying that tests were still getting run.  This situation allowed several changes to get merged to master in this period of time that broke the Windows build/test that needed to be reverted

**Comment by anitagov** [comment]
@EMumau - You need to run format-code as bors failed.

**Comment by EMumau** [comment]
bors try

**Comment by johnkord** [comment]
@EMumau I'm pretty sure errors like this ("Could not checkout trying with start point 9d925c89f97dd603788cd887672e4ca5c0e32606") happen due to a bug/design flaw in Bors. The conditions to trigger this issue are pretty easy to hit: two "bors try" runs that are kicked off within like 10 minutes of each other will cause it to occur, due to the git "trying" branch getting force pushed (overwritten) by the second "bors try" run, and the (delayed) Windows build from the first build looking for the commit hash that was associated with the first build.  The Windows build doesn't immediately start; it starts about 10 minutes after the build kicks off usually (it begins once the Linux build associated with that Windows build completes).  So if two "bors try" commands are kicked off within 10-15 minutes of each other, this issue will probably be hit, unfortunately

The best thing to do in this situation is to just run "bors try" again, but really it's not the best experience right now.  @andschwa and I have encountered this a few times in the past, and it's really quite annoying.  I should probably file an issue with them about it

**Comment by EMumau** [comment]
@johnkord : Thanks.  I was planning to ask you about that.  Does this conflict occur even if the bors try occurs on different branches?

**Comment by EMumau** [comment]
bors try

**Comment by johnkord** [comment]
@EMumau Yep. That's why it's such an obnoxious issue.  Two people might trigger a bors try within 10 minutes of each other and have no clue that they're stepping on each other. An awful experience, really

**Comment by andschwa** [comment]
I really suggest just not bothering with `bors try`. Unless you explicitly must review a change to CI configuration, `bors r+` should be sufficient. Running a build before a merge doesn't tell you much more than running a local build does. Sure, it might catch an issue with say a different compiler, but then, so will `bors r+`, and it'll kick it back to you to fix. It won't merge something that doesn't pass everything, that's the point!

**Comment by johnkord** [comment]
Pinging @soccerGB about this. Cheng-mean, I think you encountered a similar issue to the report test failure linked above, right?

**Comment by andschwa** [comment]
We've disabled the failing tests for now.

bors r+

**Pull request merged**